test/Sema/Inputs/availability_multi_other.swift

```diff
@@ -32,17 +32,18 @@ class OtherIntroduced10_51 {
     return OtherIntroduced10_52()
   }
 
-  func takes10_52(o: OtherIntroduced10_52) { 
+  func takes10_52(o: OtherIntroduced10_52) {
   }
-
+  // expected-error@-2{{'OtherIntroduced10_52' is only available on OS X 10.52 or newer}}
+  // expected-note@-3{{add @available attribute to enclosing instance method}}
+  
   @available(OSX, introduced: 10.52)
   func takes10_52Introduced10_52(o: OtherIntroduced10_52) {
   }
 
+  // expected-error@+1{{'OtherIntroduced10_52' is only available on OS X 10.52 or newer}}
   var propOf10_52: OtherIntroduced10_52 = 
```

Comment:
This would be a regression. Maybe we should skip finalizing declarations if any errors have been emitted?

Reply:
We had that, and I removed it, because it suppresses a ton of unrelated errors.

Reply:
I mean, this way we're doing unnecessary work in other files.

Reply:
We're finalizing those declarations because something needed them from the other file. In this case, they're methods of a class so we are computing the layout. Prior to my change we would skip a little work once we had produced an error, but "stop doing any work if there's any error anywhere" doesn't seem like the right default policy.

Reply:
It might be more than a little work, though—it's all structs and enums you've used throughout the file, transitively. It seems like the same reasoning to me as "don't go on to SILGen if there are errors, even though you could still get some dataflow diagnostics".

Reply:
I have an idea. Once we've encountered an error, we'll stop finalizing declarations from other source files. That way, you'll still get diagnostics in your own source file, but we won't do unnecessary work elsewhere.

Reply:
That makes a lot of sense! I forgot that finalization is now the *only* way to get certain diagnostics from decls within the current file.